dot/core/service.go

```diff
@@ -555,3 +584,22 @@ func (s *Service) GetReadProofAt(block common.Hash, keys [][]byte) (
 
 	return block, proofForKeys, nil
 }
+
+// buildExternalTransaction builds an external transaction based on the current TransactionQueueAPIVersion
+// See https://github.com/paritytech/substrate/blob/polkadot-v0.9.25/primitives/transaction-pool/src/runtime_api.rs#L25-L55
+func (s *Service) buildExternalTransaction(rt runtime.Instance, ext types.Extrinsic) (types.Extrinsic, error) {
+	runtimeVersion := rt.Version()
+	txQueueVersion := runtime.TaggedTransactionQueueVersion(runtimeVersion)
+	var externalExt types.Extrinsic
+	switch txQueueVersion {
+	case 3:
+		extrinsicParts := [][]byte{{byte(types.TxnExternal)}, ext, s.blockState.BestBlockHash().ToBytes()}
+		externalExt = types.Extrinsic(bytes.Join(extrinsicParts, nil))
+	case 2:
+		extrinsicParts := [][]byte{{byte(types.TxnExternal)}, ext}
+		externalExt = types.Extrinsic(bytes.Join(extrinsicParts, nil))
```

Comment:
nit you could take `externalExt` declaration + setting after the switch since it's common to all code branches